src/common.jl

```
@@ -59,13 +59,36 @@ function _get_injections!(
     sys::PSY.System,
 )
     sources = PSY.get_components(d -> !isa(d, PSY.ElectricLoad), PSY.StaticInjection, sys)
+    has_generator = falses(length(bus_lookup))
     for source in sources
         !PSY.get_available(source) && continue
         bus = PSY.get_bus(source)
+        # Throw an error if we see an available source at a PQ bus.
+        if PSY.get_bustype(bus) == PSY.ACBusTypes.PQ
+            throw(
+                ArgumentError(
+                    "Source $(PSY.get_name(source)) at PQ bus $(PSY.get_name(bus)). Please change the bus type to PV or REF.",
+                ),
+            )
+        end
         bus_ix = bus_lookup[PSY.get_number(bus)]
+        has_generator[bus_ix] = true
         bus_activepower_injection[bus_ix] += PSY.get_active_power(source)
         bus_reactivepower_injection[bus_ix] += PSY.get_reactive_power(source)
     end
+    # Throw an error if there's a PV/REF bus without an available source.
+    for bus in PSY.get_components(PSY.ACBus, sys)
+        bus_ix = bus_lookup[PSY.get_number(bus)]
+        if PSY.get_bustype(bus) in [PSY.ACBusTypes.PV, PSY.ACBusTypes.REF] &&
+           !has_generator[bus_ix]
```

A PV bus can also have facts devices or switched shunts. This correction needs to be more careful. Consult with @mcllerena on the behavior of these other components.

---

Messasing Marck, PSSE expects a PV bus to satisfy: (has available generator) AND NOT (has unlocked switched shunt/FACTS device). If that is violated, PSSE outputs a warning, but oddly it still runs the powerflow. Should we do the same, or should we throw an error?

---

Throw and error.